pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -102,8 +102,12 @@ public static boolean isStarTreeDisabled(QueryContext queryContext) {
           queue.addAll(filterNode.getChildren());
           break;
         case OR:
-          // Star-tree does not support OR filter
-          return null;
+          if (isOrClauseValidForStarTree(filter)) {
```

Comment:
We cannot mix the predicates under OR with predicates under AND. One example filter would be `a < 10 AND (b < 0 OR b > 10)`, where we need to preserve the tree structure